### DISCUSSION TITLE
Dart Codegen

### DISCUSSION THREAD

**Discussion by cybertheory** [apple/pkl#109] [Ideas]
Hi! Been working with Dart and Flutter for a few years now. I am down to get started and contribute Dart code-gen capabilities.

**Comment by ensaremirerol** [comment]
Hey, if you are up for it. We can start working on it together?

**Comment by bioball** [comment]
Sounds great! We just published some documentation about how to interact with Pkl from a host application: https://pkl-lang.org/main/current/bindings-specification/index.html

For codegen: we might provide some helper tools, but for now, the easiest way to do this is to copy/paste the code from either [pkl-go](https://github.com/apple/pkl-go/tree/main/codegen), or [pkl-swift](https://github.com/apple/pkl-swift/tree/main/codegen).

**Comment by yildirayyorulmaz** [comment]
Could you provide an update on the progress?  I can give you a hand

**Reply by ensaremirerol** [reply]
Well there is nothing to share, no has followed on it. Though if you are interested we can start developing.

**Reply by nikeokoronkwo** [reply]
Is there any progress on this so far?

**Reply by ensaremirerol** [reply]
Well, I think everyone is waiting each other on this. Maybe I can start it and we can build on top of that

**Reply by nikeokoronkwo** [reply]
I'd be very willing to join to work on this. Let me know when you've started

**Reply by bioball** [reply]
Go for it! And feel free reach out to us if you have any questions about this!

**Reply by savekirk** [reply]
I have started working on this. You can check the [project](https://github.com/savekirk/pkl-dart) for more details.

**Reply by yildirayyorulmaz** [reply]
I'll check this out. Thank you @savekirk
